src/korma/core.clj

```diff
@@ -743,25 +767,36 @@
   (let [sub-ent (assoc-db-to-entity query sub-ent)
         [ent-key sub-ent-key] (get-join-keys rel (:ent query) sub-ent)]
     (post-query query
-                (partial map
-                         (fn [ent]
-                           (merge-with-unique-keys (get-key-naming-strategy query)
-                                                   ent
-                                                   (first
-                                                     (select sub-ent
-                                                             (body-fn)
-                                                             (where {sub-ent-key (get ent ent-key)})))))))))
+      (partial map
+         (fn [ent]
+           (merge-with-unique-keys (get-key-naming-strategy query)
+              ent
+              (first
+                (select sub-ent
+                  (body-fn)
+                  (where
+                    (if (seq? ent-key)
+                      (let [ent-kval (map #(get ent %) ent-key)]
+                        (apply hash-map
+                           (interleave sub-ent-key ent-kval)))
+                      {sub-ent-key (get ent ent-key)}))))))))))
 
 (defn- with-one-to-one-now [rel query sub-ent body-fn]
-  (let [table (if (:alias rel) [(:table sub-ent) (:alias sub-ent)] (:table sub-ent))
-        [ent-key sub-ent-key] (get-join-keys rel (:ent query) sub-ent)]
+  (let [ent (:ent query)
+        table (if (:alias rel) [(:table sub-ent) (:alias sub-ent)] (:table sub-ent))
+        [ent-key sub-ent-key] (get-join-keys rel ent sub-ent)]
     (bind-query query
                 (merge-query
                   (make-sub-query sub-ent body-fn)
                   (join query
                         table
-                        (= (raw (eng/prefix sub-ent sub-ent-key))
-                           (raw (eng/prefix (:ent query) ent-key))))))))
+                        (if (seq? ent-key) 
```

Comment:
implemented composite key support here by handling seq-keys differently; this prevents existing tests from breaking by leaving handling for single pk/fk keys unmodified.